src/lib.rs

```diff
@@ -514,6 +514,37 @@ impl Client {
         self.issue(WebDriverCommand::ExecuteScript(cmd)).await
     }
 
+    /// Execute the given async JavaScript script in the current browser session.
+    ///
+    /// `args` is available to the script inside the `arguments` array. Since `Element` implements
+    /// `Serialize`, you can also provide serialized `Element`s as arguments, and they will
+    /// correctly deserialize to DOM elements on the other side.
+    ///
+    /// The final arg in `args` is a callback function that when called will trigger completion
+    /// with the provided value. For example,
+    ///
+    /// ```javascript
+    /// const done = arguments.pop();
+    ///
+    /// setInterval(() => {
+    ///     // ... do stuff
+    ///     done(valueToReturn);
+    /// }, 5000);
+    /// ```
```

Comment:
I don't quite follow this example — could you include the Rust part of the example to help illustrate the callback mechanism for the last argument?

Reply:
Added some changes here which I hope makes it clearer: https://github.com/jonhoo/fantoccini/pull/107/commits/b210632ba868dc867a4cec3b7af849a828910632

Reply:
Ah, I think what was missing to make this click for me was that this function automatically makes the arguments available as the `arguments` variable in the JS code. I wonder if we can make this more explicit with some language along the lines of:
> The provided JavaScript has access to the provided arguments through the `arguments` array. The array also holds an additional element at the end that provides a completion callback for the asynchronous code. [..]

Reply:
Hmm, GitHub is acting oddly for me, still showing the prior docs in the review. How does this look?

```
    /// Execute the given async JavaScript script in the current browser session.
    ///
    /// The provided JavaScript has access to the provided `args` through the `arguments` array. The array also 
    /// holds an additional element at the end that provides a completion callback for the asynchronous code.
    ///
    /// Since `Element` implements `Serialize`, you can also provide serialized `Element`s as arguments, and they will
    /// correctly deserialize to DOM elements on the other side.
    ///
    /// # Examples
    ///
    /// Call a web API from the browser and retrieve the value asynchronously
    ///
    /// ```ignore
    /// const JS: &'static str = r#"
    ///     const [date, callback] = arguments;
    ///
    ///     fetch(`http://weather.api/${date}/hourly`)
    ///     // whenever the HTTP Request completes,
    ///     // send the value back to the Rust context
    ///     .then(data => {
    ///         callback(data.json())
    ///     })
    /// "#;
    ///
    /// let weather = client.execute_async(JS, vec![date]).await?;
    /// ```
```

Reply:
Yeah, that looks much better. Thinking some more about it, I think we want to be _even_ more explicit:
> The provided JavaScript has access to `args` through the JavaScript variable `arguments`.

Reply:
Updated again, thanks!